modules/auth/repo_form.go

```diff
@@ -302,6 +303,32 @@ func (f *ReactionForm) Validate(ctx *macaron.Context, errs binding.Errors) bindi
 	return validate(errs, ctx.Data, f, ctx.Locale)
 }
 
+// IssueLockForm form for locking an issue
+type IssueLockForm struct {
+	Reason string `binding:"Required"`
+}
+
+// Validate validates the fields
+func (i *IssueLockForm) Validate(ctx *macaron.Context, errs binding.Errors) binding.Errors {
+	return validate(errs, ctx.Data, i, ctx.Locale)
+}
+
+// HasValidReason checks to make sure that the reason submitted in
+// the form matches any of the values in the config
+func (i IssueLockForm) HasValidReason() bool {
```

Comment:
Maybe add this to `Validate`.

Reply:
I think it is better if explicit enough.. And that regular `Validate` is just meant for validating the struct binding..

Reply:
Thoughts @JonasFranzDEV ?

Reply:
Nudge

Reply:
ping @jonasfranz